FrontEnd/src/components/profileList/ProfileCard.jsx

```diff
@@ -0,0 +1,145 @@
+    activities: !data.activities.length ? null : data.activities,
+    region: regions.find((region) => region.key == data.region).value,
+    address: data.address,
+    categories: !data.categories.length ? null : data.categories,
+    isSaved: isSaved,
+    commonInfo: data.common_info,
+  };
+
+  const addressLine = `${profile.region ? profile.region : ""}`;
+  const activitiesLine = profile.activities.map((activity) => activity.name).join(", ");
+
+  const filledStar = (
+    <StarFilled
+      style={{ color: "#FFD800", fontSize: "24px" }}
+      onClick={onStarClick}
+    />
+  );
+  const outlinedStar = (
+    <StarOutlined
+      style={{ color: "#FFD800", fontSize: "24px" }}
+      onClick={onStarClick}
+    />
+  );
+
+  const CategoryBadges = ({ categories }) => {
+    return (
+      <>
+        {categories
+          ? categories.map((category) => (
+              <Badge
+                key={category.id}
+                size="medium"
+                count={category.name.toUpperCase()}
+                style={{
+                  backgroundColor: "#1F9A7C",
+                  fontWeight: 600,
+                  fontFamily: "Inter",
+                  fontSize: 10,
+                }}
+              />
+            ))
+          : ""}
+      </>
+    );
+  };
+
+  // FIXME: fix race condition when switching from saved to all w/saved changes
+  function onStarClick() {
+    axios({
+      method: "post",
+      url: `${process.env.REACT_APP_BASE_API_URL}/api/saved-list/`,
+      withCredentials: true,
+      data: {
+        company_pk: profile.id,
+      },
+    })
+      .then(() => setIsSaved(!isSaved))
+      .catch((error) => console.log(error));
+  }
+  // TODO: add logo from db once it's implemented on the server side
+
+  return (
+    <div className={css["company-card"]}>
+      <div className={css["logo-box"]}>
+        <img
+          className={css.logo}
+          src={`${process.env.PUBLIC_URL}/companies-logos/1.png`}
+        />
+      </div>
+      <div className={css.content}>
+        <div className={css["content-header"]}>
+          <div className={css["content-header__activity"]}>
+            <p className={css["content-header__activity--text"]}>
+              {activitiesLine}
+            </p>
+          </div>
+          <div className={css["content-header__name"]}>
+            {profile.companyName}
+          </div>
+          <div className={css["content-header__address"]}>{addressLine}</div>
+        </div>
+        <div className={css["content__common-info"]}>
+          <Paragraph ellipsis={{ rows: 3, expandable: false }}>
+            {profile.commonInfo}
+          </Paragraph>
+        </div>
+        <div className={css["content__categories"]}>
+          <CategoryBadges categories={profile.categories} />
+        </div>
+      </div>
+      {isAuthorized ? (profile.isSaved ? filledStar : outlinedStar) : null}
+    </div>
+  );
+}
+
+// FIXME: add null acceptance for optional fields
```

Comment:
I thing nothing blocks us from resolving it

Reply:
that's funny. i meant that i needed to specify the required ones :D either way done